egov-pg-service/src/main/java/org/egov/pg/web/controllers/TransactionsApiController.java

```diff
@@ -104,11 +109,17 @@ public TransactionsApiController(TransactionService transactionService, GatewayS
      * @return list of active gateways that can be used for payments
      */
     @RequestMapping(value = "/gateway/v1/_search", method = RequestMethod.POST)
-    public ResponseEntity<Set<String>> transactionsV1AvailableGatewaysPost() {
+    public ResponseEntity<List> transactionsV1AvailableGatewaysPost(@RequestBody TransactionRequest transactionRequest) throws Exception {
+        Transaction transaction = transactionRequest.getTransaction();
+        RequestInfo requestInfo = transactionRequest.getRequestInfo();
+        String tenantId = transaction.getTenantId();
+
+        List listOfGateway = gatewayMetadata.listOfGateways(requestInfo,tenantId);
 
-        Set<String> gateways = gatewayService.getActiveGateways();
-        log.debug("Available gateways : " + gateways);
-        return new ResponseEntity<>(gateways, HttpStatus.OK);
+        return new ResponseEntity<>(listOfGateway,HttpStatus.OK);
```

Comment:
Is this using the default format we use in other APIs with `ResponseInfo` and a fields for the actual service response?

Reply:
I am unable to return it in the generic format, can you fix this for me please?